scripts/us_census/pep/monthly_population_estimate/download.py

```diff
@@ -0,0 +1,226 @@
+    cols = [
+        "Year and Month", "Resident Population",
+        "Resident Population Plus Armed Forces Overseas", "Civilian Population",
+        "Civilian NonInstitutionalized Population"
+    ]
+    df.columns = cols
+    for col in df.columns:
+        df[col] = df[col].str.replace(",", "")
+    return df
+
+
+def _clean_txt_file(df: pd.DataFrame,
+                    scaling_factor_txt_file: int) -> pd.DataFrame:
+    """
+    This method cleans the dataframe loaded from a txt file format.
+    Also, Performs transformations on the data.
+
+    Arguments:
+        df (DataFrame) : DataFrame of txt dataset
+        scaling_factor_txt_file (int) : Scaling factor for text file
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for txt dataset.
+    """
+    # Month and Year are concatenated into a single column if they are not None
+    df['Year and Month'] = df[['Year and Month', 'Date']]\
+                                    .apply(_sum_cols, axis=1)
+    df.drop(columns=['Date'], inplace=True)
+    for col in df.columns:
+        df[col] = df[col].str.replace(",", "")
+    idx = df[df['Resident Population'] == "(census)"].index
+
+    # The index numbers alotted as per where the columns are present to
+    # move the columns left
+    resident_population = 1
+    resident_population_plus_armed_forces_overseas = 2
+    civilian_population = 3
+    civilian_noninstitutionalized_population = 4
+
+    # Moving the row data left upto one index value.
+    df.iloc[idx, resident_population] = df.iloc[idx][
+        "Resident Population Plus Armed Forces Overseas"]
+    df.iloc[idx, resident_population_plus_armed_forces_overseas] = df.iloc[idx][
+        "Civilian Population"]
+    df.iloc[idx, civilian_population] = df.iloc[idx][
+        "Civilian NonInstitutionalized Population"]
+    df.iloc[idx, civilian_noninstitutionalized_population] = np.NAN
+
+    # Multiplying the data with scaling factor 1000.
+    for col in df.columns:
+        if "year" not in col.lower():
+            if scaling_factor_txt_file != 1:
+                df[col] = df[col].apply(_mulitply_scaling_factor,
+                                        scaling_factor=scaling_factor_txt_file)
+    return df
+
+
+def _download(download_path: str, file_urls: list) -> None:
+    """
+    This method iterates on each url and calls the above defined
+    functions to download and clean the data.
+
+    Arguments:
+        download_path (str) : Local Path to download datasets from URLS
+        file_urls (list) : List of dataset URLS.
+
+    Returns:
+        df (DataFrame) : Transformed DataFrame for txt dataset.
+    """
+    if not os.path.exists(download_path):
+        os.mkdir(download_path)
+    for url in file_urls:
+        _save_data(url, download_path)
+
+
+def main(_):
+    file_urls = _FLAGS.us_census_pep_monthly_pop_estimate_url
+    path = os.path.dirname(os.path.abspath(__file__)) + os.sep + "input_data"
+    _download(path, file_urls)
```

Comment:
Given there is custom transformation logic in this can we add a test case for this?

I believe you can pass `file://localhost/path/to/table.csv` to read_csv, read_excel instead of http urls in test mode.

Reply:
We have added the test logic which would use download.py using csv from local as suggested.